src/code/code_0x8012EC80.c

```diff
@@ -0,0 +1,406 @@
+        SCENE_TAKARAYA,     SCENE_YADOYA,       SCENE_SONCHONOIE,
+        SCENE_MILK_BAR,     SCENE_ICHIBA,       SCENE_BOMYA,
+        SCENE_AYASHIISHOP,  SCENE_8ITEMSHOP,    SCENE_DOUJOU,
+        SCENE_POSTHOUSE,    SCENE_TAKARAKUJI,   SCENE_BACKTOWN,
+        SCENE_CLOCKTOWER,   SCENE_INSIDETOWER,  SCENE_OKUJOU,
+        SCENE_ALLEY,        SCENE_DEKUTES,      -1,
+    },
+    // Snowhead
+    {
+        SCENE_10YUKIYAMANOMURA,
+        SCENE_10YUKIYAMANOMURA2,
+        SCENE_KAJIYA,
+        SCENE_12HAKUGINMAE,
+        SCENE_14YUKIDAMANOMITI,
+        SCENE_GORONRACE,
+        SCENE_GORON_HAKA,
+        SCENE_17SETUGEN,
+        SCENE_17SETUGEN2,
+        -1,
+    },
+    // Ikana Graveyard
+    {
+        SCENE_BOTI,
+        SCENE_DANPEI2TEST,
+        -1,
+    },
+    // Ikana Canyon
+    {
+        SCENE_CASTLE,
+        SCENE_IKNINSIDE,
+        SCENE_IKANA,
+        SCENE_SECOM,
+        SCENE_MUSICHOUSE,
+        SCENE_RANDOM,
+        SCENE_REDEAD,
+        SCENE_TOUGITES,
+        SCENE_HAKASHITA,
+        -1,
+    },
+    // Goron Village
+    {
+        SCENE_11GORONNOSATO,
+        SCENE_11GORONNOSATO2,
+        SCENE_16GORON_HOUSE,
+        SCENE_GORONSHOP,
+        -1,
+    },
+    // Stone Tower
+    {
+        SCENE_F40,
+        SCENE_F41,
+        -1,
+    },
+};
+
+s32 func_8012EC80(GlobalContext* globalCtx) {
+    if (gSaveContext.buttonStatus[0] == BTN_DISABLED) {
+        return 0xFF;
+    } else if (gSaveContext.unk_1015 == 0xFF) {
+        return 0xFF;
+    } else if (gSaveContext.equips.buttonItems[((gSaveContext.playerForm == 4) ? 0 : gSaveContext.playerForm)][0] ==
+               0xFF) {
+        if (globalCtx->interfaceCtx.unk_21C != 0) {
+            if (globalCtx->interfaceCtx.unk_21E != 0) {
+                return globalCtx->interfaceCtx.unk_21E;
+            }
+        }
+        return 0xFF;
+    } else {
+        return gSaveContext.equips.buttonItems[((gSaveContext.playerForm == 4) ? 0 : gSaveContext.playerForm)][0];
+    }
+}
+
+void func_8012ED34(s16 equipment);
+#pragma GLOBAL_ASM("./asm/non_matchings/code/code_0x8012EC80/func_8012ED34.asm")
+
+u8 func_8012ED78(GlobalContext* globalCtx, s16 equipment);
+#pragma GLOBAL_ASM("./asm/non_matchings/code/code_0x8012EC80/func_8012ED78.asm")
+
+void func_8012EDE8(s16 arg0, u32 arg1);
+#pragma GLOBAL_ASM("./asm/non_matchings/code/code_0x8012EC80/func_8012EDE8.asm")
+
+s32 func_8012EE34(s16 arg0) {
+    s16 num = 0;
+
+    if (arg0 >= 0x20) {
+        if (arg0 < 0x40) {
+            num = 1;
+        } else if (arg0 < 0x60) {
+            num = 2;
+        } else if (arg0 < 0x80) {
+            num = 3;
+        } else if (arg0 < 0xA0) {
+            num = 4;
+        } else if (arg0 < 0xC0) {
+            num = 5;
+        } else if (arg0 < 0xE0) {
+            num = 6;
+        }
+    }
+
+    if (gSaveContext.roomInf[125][num] & gBitFlags[arg0 - (num << 5)]) {
+        return 1;
+    }
+
+    return 0;
+}
+
+/* static */ u16 D_801C2380[12][6] = {
```

Comment:
I guess MM didn't get the new asm processor?

Reply:
nope

Reply:
Why not try updating the submodule?